### DISCUSSION TITLE
Creating an API on MicroPython using Microdot

### DISCUSSION THREAD

**Discussion by beyonlo** [miguelgrinberg/microdot#234] [General]
Hello

Is it completely possible to create a complete API (REST API) using Microdot? I mean, in Python the most common frameworks used to create APIs are Flask and FastAPI. I would like to create an API to serve in MicroControllers, so I need to use MicroPython. So I would like to use Microdot on MicroPython to serve a complete API.

Thank you in advance!

**Comment by miguelgrinberg** [comment]
Well, yes. An API is just a collection of HTTP endpoints.

**Reply by beyonlo** [reply]
Is there any simple example for that?

Thank you!

**Reply by miguelgrinberg** [reply]
No. I need to add API examples, I now realize I don't have any. I will add one: https://github.com/miguelgrinberg/microdot/issues/235

**Reply by beyonlo** [reply]
Thank you very much!
